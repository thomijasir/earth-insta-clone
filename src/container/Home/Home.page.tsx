import React, { FC, useState, useEffect, useContext } from 'react';
import { AppContext } from '../../store/AppProvider';
import { TITLE_HOME_PAGE } from '../../constants';
import reactLogo from '../../assets/images/react.svg';
import './Home.scss';
import HeadBar from '../../components/HeadBar/HeadBar';

export interface IProps {}

const Home: FC<IProps> = () => {
  const context = useContext(AppContext);
  const [title] = useState('React Boilerplate');

  useEffect(() => {
    console.log(TITLE_HOME_PAGE);
    console.log('HOME CONTEXT: ', context);
  }, []);

  return (
    <div className="home-page">
      <HeadBar />
    </div>
  );
};

export default Home;
