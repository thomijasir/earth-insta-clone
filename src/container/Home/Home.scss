.home-page {
  height: 100vh;
  display: flex;
  flex-flow: column;
  box-sizing: border-box;
  overflow: hidden;
  justify-content: space-between;
  .col-home {
    &.home-group {
      overflow-y: auto;
      &::-webkit-scrollbar {
        display: none;
      }
    }
  }
}
