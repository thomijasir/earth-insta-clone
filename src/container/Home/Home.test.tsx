import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import render from 'react-test-renderer';
import Home from './Home.page';

afterEach(() => {
  jest.clearAllMocks();
});

beforeEach(() => {
  // MOCK MATH RANDOM
  jest.spyOn(global.Math, 'random').mockReturnValue(1);
});

describe('HomePage Container Snap Test', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = render.create(
        <BrowserRouter>
          <Home />
        </BrowserRouter>,
      );
      expect(wrapper.toJSON()).toMatchSnapshot();
    });
  });
});
