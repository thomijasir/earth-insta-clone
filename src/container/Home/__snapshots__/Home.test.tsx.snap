// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage Container Snap Test render() renders the component 1`] = `
<div
  className="home-page"
>
  <div
    className="head-bar-comp"
  >
    <div
      className="col-head-bar insta-logo"
    >
      <img
        alt="Intagram Logo Text"
        src="instagram-text.jpg"
      />
    </div>
    <div
      className="col-head-bar head-icon"
    >
      <img
        alt="Intagram Logo Text"
        src="plus-icon.jpg"
      />
    </div>
    <div
      className="col-head-bar head-icon"
    >
      <img
        alt="Intagram Logo Text"
        src="heart-icon.jpg"
      />
    </div>
    <div
      className="col-head-bar head-icon"
    >
      <img
        alt="Intagram Logo Text"
        src="comment-icon.jpg"
      />
    </div>
  </div>
</div>
`;
